backends/guest/guest_svc_generate.c

```diff
@@ -493,14 +506,19 @@ static int parse_options(int key, char *arg, struct argp_state *state)
 		/* else set input and output formats */
 		args->input_form = strtok(arg, ":");
 		args->output_form = strtok(NULL, ":");
+
+		/* verify both input and output forms are parsed correctly, error otherwise */
+		if (!args->input_form || !args->output_form) {
+			prlog(PR_ERR,
+			      "ERROR: '%s' is not in the correct '<input_format>:<output_format>' form, see usage...\n",
+			      arg);
+			rc = ARG_PARSE_FAIL;
+		}
 		break;
 	case ARGP_KEY_SUCCESS:
 		/* check that all essential args are given and valid */
 		if (args->help_flag)
 			break;
-		else if (args->input_form == NULL || args->output_form == NULL)
-			prlog(PR_ERR, "ERROR: incorrect '<input_format>:<output_format>', see "
-				      "usage...\n");
```

Comment:
This final check is necessary otherwise the user can get away with not setting the `in:out` argument:
```
$ ./bin/secvarctl-dbg -m guest generate -i sbat_in -o sbat.auth 
AddressSanitizer:DEADLYSIGNAL
=================================================================
==566168==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x0000004377a9 bp 0x7ffe8abfeae0 sp 0x7ffe8abfeab0 T0)
==566168==The signal is caused by a READ memory access.
==566168==Hint: address points to the zero page.
    #0 0x4377a8 in parse_options backends/guest/guest_svc_generate.c:525
    #1 0x7fe995198892 in argp_parse (/lib64/libc.so.6+0x13a892)
    #2 0x438dbc in guest_generate_command backends/guest/guest_svc_generate.c:666
    #3 0x4069b4 in main /home/nchild/IBM/secvarctl-stuff/erichte-guest-devel/secvarctl/secvarctl.c:250
    #4 0x7fe995098d84 in __libc_start_main (/lib64/libc.so.6+0x3ad84)
    #5 0x4031dd in _start (/home/nchild/IBM/secvarctl-stuff/erichte-guest-devel/secvarctl/bin/secvarctl-dbg+0x4031dd)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV backends/guest/guest_svc_generate.c:525 in parse_options
==566168==ABORTING
```
This can avoided by adding back this conditional:
```
$ ./bin/secvarctl-dbg -m guest generate -i sbat_in -o sbat.auth 
ERROR: incorrect '<input_format>:<output_format>', see usage...
Usage: secvarctl -m guest generate [OPTION...]
            <input_format>:<output_format> -i <input_file> -o <output_file>
  or:  secvarctl -m guest generate [OPTION...]
            reset -i <input_file> -o <output_file>
Try `secvarctl -m guest generate --help' or `secvarctl -m guest generate
--usage' for more information.
failed during argument parsing
RESULT: FAILURE
```

Reply:
Yep, that's an oversight on my part. Re-added the check, and tweaked the message to include potentially missing the argument.